client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js

```diff
@@ -1,172 +1,84 @@
-import ThemeTierTooltipTracker from './theme-tier-tooltip-tracker';
+import ThemeTierPlanUpgradeBadge from './theme-tier-upgrade-badge';
+
+const commonBadgeProps = {
+	focusOnShow: false,
+	isClickable: false,
+	shouldHideTooltip: true,
+	shouldHideIcon: false,
+};
 
-export default function ThemeTierPartnerBadge() {
+export default function ThemeTierPartnerBadge( {
+	showPartnerPrice,
+	hideBackgroundOnUpgrade,
+	isLongLabel,
+} ) {
 	const translate = useTranslate();
 	const siteId = useSelector( getSelectedSiteId );
-	const planSlug = useSelector( ( state ) => getSitePlanSlug( state, siteId ) ?? '' );
-
-	const isEcommerceTrialMonthly = planSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY;
-	const planTooltipName = isEcommerceTrialMonthly ? 'ecommerce' : 'business';
+	const { themeId } = useThemeTierBadgeContext();
 
-	const { showUpgradeBadge, themeId } = useThemeTierBadgeContext();
-	const isPartnerThemePurchased = useSelector( ( state ) =>
-		siteId ? isMarketplaceThemeSubscribed( state, themeId, siteId ) : false
+	const isPartnerThemePurchased = useSelector(
+		( state ) => ( siteId ? isMarketplaceThemeSubscribed( state, themeId, siteId ) : false ),
+		( prev, next ) => prev === next
 	);
-	const subscriptionPrices = useSelector( ( state ) =>
-		getMarketplaceThemeSubscriptionPrices( state, themeId )
+
+	const subscriptionPrices = useSelector(
+		( state ) => getMarketplaceThemeSubscriptionPrices( state, themeId ),
+		( prev, next ) => prev.month === next.month && prev.year === next.year
 	);
-	const isThemeAllowed = useIsThemeAllowedOnSite( siteId, themeId );
 
-	const labelText = isThemeAllowed
-		? translate( 'Subscribe' )
-		: translate( 'Upgrade and Subscribe' );
+	const isThemeAllowed = useIsThemeAllowedOnSite( siteId, themeId );
 
-	const getTooltipMessage = () => {
-		if ( isPartnerThemePurchased && ! isThemeAllowed && ! isEcommerceTrialMonthly ) {
-			return createInterpolateElement(
-				translate(
-					'You have a subscription for this theme, but it will only be usable if you have the <link>%(businessPlanName)s plan</link> or higher on your site.',
-					{ args: { businessPlanName: getPlan( PLAN_BUSINESS )?.getTitle() ?? '' } }
-				),
-				{
-					Link: <ThemeTierBadgeCheckoutLink plan={ planTooltipName } />,
-				}
+	const priceBadge = useMemo( () => {
+		if ( ! isThemeAllowed ) {
+			return (
+				<ThemeTierPlanUpgradeBadge
+					showPartnerPrice={ showPartnerPrice }
+					hideBackgroundOnUpgrade={ hideBackgroundOnUpgrade }
+				/>
 			);
 		}
 
-		if ( isPartnerThemePurchased && ! isThemeAllowed && isEcommerceTrialMonthly ) {
-			return createInterpolateElement(
-				translate(
-					"You have a subscription for this theme, but it isn't usable on a trial plan site. Please upgrade to the <link>%(ecommercePlanName)s plan</link> to install this theme.",
-					{ args: { ecommercePlanName: getPlan( PLAN_ECOMMERCE )?.getTitle() ?? '' } }
-				),
-				{
-					Link: <ThemeTierBadgeCheckoutLink plan={ planTooltipName } />,
-				}
-			);
-		}
```

Comment:
Same for partner themes and Commerce trial sites. We cannot ask them to buy the Business plan, because their only option is the Commerce plan:

Before | After
--- | ---
<img width="444" alt="Screenshot 2025-02-19 at 19 59 11" src="https://github.com/user-attachments/assets/f7bfb919-3429-416f-bc85-48320e20fa38" />  | <img width="387" alt="Screenshot 2025-02-19 at 19 59 18" src="https://github.com/user-attachments/assets/c21cea9d-faab-4fa5-afc7-fe57ad0d25b6" />